lib/pharos/phases/validate_version.rb

```diff
@@ -60,9 +61,19 @@ def kubeconfig
 
       # @return [K8s::Resource, nil]
       def previous_config_map
+        original_ssl_verify_peer = kube_client.transport.options[:ssl_verify_peer]
+
         kube_client.api('v1').resource('configmaps', namespace: 'kube-system').get('pharos-config')
+      rescue Excon::Error::Socket => ex
+        raise if !kube_client.transport.options[:ssl_verify_peer] # don't retry if ssl verify is false
+
+        kube_client.transport.options[:ssl_verify_peer] = false
+        logger.debug { "Encountered #{ex.class.name} : #{ex.message} - retrying with ssl_verify_peer = false" }
```

Comment:
Maybe output something with info level so that user realises that something is a bit off?